src/Aspire.Hosting/Postgres/PostgresBuilderExtensions.cs

```diff
@@ -62,6 +64,22 @@ public static IResourceBuilder<PostgresDatabaseResource> AddDatabase(this IResou
                                          .WithManifestPublishingCallback(context => WritePostgresDatabaseToManifest(context, postgresDatabase));
     }
 
+    /// <summary>
+    /// Adds a pgAdmin 4 administration and development platform for PostgreSQL to the application model.
+    /// </summary>
+    /// <param name="builder">The PostgreSQL server resource builder.</param>
+    /// <param name="port">The host port for the application ui</param>
+    /// <param name="options"><see cref="PgAdminOptions"/> for the container (optional).</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{PostgresContainerResource}"/>.</returns>
+    public static IResourceBuilder<PostgresContainerResource> WithPgAdmin(this IResourceBuilder<PostgresContainerResource> builder, int? port, PgAdminOptions? options = null)
```

Comment:
Why just not just add the resource here directly?

Reply:
I wanted to prevent multiple calls to WithPgAdmin from resulting in the error that a "pgAdmin" resource already exists.

I will add the container resource to WithPgAdmin itself and leave it up to the developer to ensure that it is only used once.

I'll add passing an optional resource name so the developer has control over this as well.

Reply:
Done